### PR TITLE
add recipe controller unit tests

### PR DESCRIPTION
**Pull request by boraykasap**

### CODE REVIEW

**Review by EnesBaserr — COMMENTED** [review]
We are going to merge when Recipe related responses finalized.

File: backend/src/test/java/com/group1/cuisines/RecipeControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        ErrorResponse errorResponse = (ErrorResponse) response.getBody();
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assert errorResponse != null;
+        assertEquals(200, errorResponse.getStatus());
+    }
+
+    @Test
+    void testGetRecipes() {
+        List<RecipeDto> recipes = Collections.singletonList(new RecipeDto());
```
Comment (atakanyasar) [inline backend/src/test/java/com/group1/cuisines/RecipeControllerTest.java:69]: RecipeDto is removed.

File: backend/src/test/java/com/group1/cuisines/RecipeControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        assertEquals("Authentication required", errorResponse.getMessage());
+    }
+
+    @Test
+    void testCreateRecipe_Success() throws Exception {
+        User user = User.builder().username("testUser").build();
+        when(authenticationService.getUser()).thenReturn(Optional.of(user));
+
+        NewRecipeDto newRecipe = new NewRecipeDto();
+        RecipeDetailDto recipeDetails = new RecipeDetailDto();
```
Comment (atakanyasar) [inline backend/src/test/java/com/group1/cuisines/RecipeControllerTest.java:101]: RecipeDetailDto is removed

**Review by atakanyasar — CHANGES_REQUESTED** [review]

**Comment by mmtftr** [comment]
closing due to new term

**Pull request closed without merging**